ui/app/components/calendar-widget.js

```diff
@@ -92,35 +79,36 @@ class CalendarWidget extends Component {
         // if they are on the view where the start year equals the display year, check which months should not show.
         let startMonth = this.args.startTimeDisplay.split(' ')[0]; // returns month name e.g. January
         // return the index of the startMonth
-        let startMonthIndex = this.arrayOfMonths.indexOf(startMonth);
+        let startMonthIndex = this.args.arrayOfMonths.indexOf(startMonth);
         // then add readOnly class to any month less than the startMonth index.
         if (startMonthIndex > elementMonthId) {
           e.classList.add('is-readOnly');
         }
       }
+      // Compare values so the user cannot select an endTime after the endTime returned from counters/activity response on page load.
+      // ARG TODO will need to test if no data is returned on page load.
+      if (this.displayYear.toString() === this.args.endTimeFromResponse.split(' ')[1]) {
+        let endMonth = this.args.endTimeFromResponse.split(' ')[0];
+        let endMonthIndex = this.args.arrayOfMonths.indexOf(endMonth);
+        // add readOnly class to any month that is older (higher) than the endMonth index. (e.g. if nov is the endMonth of the endTimeDisplay, then 11 and 12 should not be displayed 10 < 11 and 10 < 12.)
+        if (endMonthIndex < elementMonthId) {
+          e.classList.add('is-readOnly');
+        }
+      }
     });
   }
 
   @action
-  selectCurrentBillingPeriod() {
-    // ARG TOOD send to dashboard the select current billing period. The parent may know this it's just a boolean.
-    // Turn the calendars off if they are showing.
+  selectCurrentBillingPeriod(D) {
+    this.args.handleCurrentBillingPeriod(); // resets the billing startTime and endTime to what it is on init via the parent.
     this.showCalendar = false;
-    this.showSingleMonth = false;
+    D.actions.close(); // close the dropdown.
   }
   @action
-  selectEndMonth(month, year, element) {
-    this.addClass(element.target, 'is-selected');
+  selectEndMonth(month, year, D) {
     this.toggleShowCalendar();
     this.args.handleClientActivityQuery(month, year, 'endTime');
-  }
-
-  @action
```

Comment:
All things singleMonth were removed because this is not something we're now doing in 1.10